### PR TITLE
Update etcd makefile to build 3.4.9 image

### PR DESCRIPTION
**Pull request by jingyih**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We are updating etcd client version to 3.4.9 in #92075. This PR updates the etcd makefile to match the version, so that 3.4.9 images can be built.

**Which issue(s) this PR fixes**:
#91266

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig api-machinery
/area etcd

### CODE REVIEW

**Comment by jingyih** [comment]
cc @dims @jpbetz @wenjiaswe

**Comment by jingyih** [comment]
@dims `pull-kubernetes-dependencies` is complaining about etcd version mismatch in the codebase. This means we cannot modify the etcd makefile to trigger the automatic image building process as the first step?

**Comment by BenTheElder** [comment]
if the build is now automated and done in postsubmit this entry should be forked into the image and it's consumers https://github.com/kubernetes/kubernetes/blob/2402bfd4bc329ae755250f5214351b1408a48eab/build/dependencies.yaml#L46

**Comment by jingyih** [comment]
> if the build is now automated and done in postsubmit this entry should be forked into the image and it's consumers
> 
> https://github.com/kubernetes/kubernetes/blob/2402bfd4bc329ae755250f5214351b1408a48eab/build/dependencies.yaml#L46

@BenTheElder Could you help take a look if this is right? 
 https://github.com/kubernetes/kubernetes/pull/92078/commits/3ba87cd6276ce70abc3567123804d27ceaec1492

File: build/dependencies.yaml [hunk 1]
```diff
@@ -64,6 +60,14 @@ dependencies:
     - path: test/e2e/framework/nodes_util.go
       match: const etcdImage
 
+  - name: "etcd"
```
Comment (BenTheElder) [inline build/dependencies.yaml:63]: maybe give these entries different names?

```suggestion
  - name: "etcd-image"
```

**Comment by BenTheElder** [comment]
one suggestion, otherwise LGTM, thank you!

**Comment by justaugustus** [comment]
FYI @kubernetes/release-engineering

File: build/dependencies.yaml (hunk 1 above)
Comment (jingyih) [reply build/dependencies.yaml]: Done. Thanks!

**Comment by BenTheElder** [comment]
finally green :crossed_fingers:

**Comment by BenTheElder** [comment]
@jingyih did we bump the consumers?

**Comment by jingyih** [comment]
> @jingyih did we bump the consumers?

Sorry for the delay. Just created https://github.com/kubernetes/k8s.io/pull/973 to promote the images to prod repo. Once it is done, we can bump the consumers.

**Pull request merged**